lib/kernel/cuda/get_global_id.c

```diff
@@ -0,0 +1,51 @@
+/* OpenCL built-in library: get_global_id() for CUDA
+
+   Copyright (c) 2016 James Price
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to deal
+   in the Software without restriction, including without limitation the rights
+   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+   copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in
+   all copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+   THE SOFTWARE.
+*/
+
+extern uint _local_size_x;
+extern uint _local_size_y;
+extern uint _local_size_z;
+
+extern uint _group_id_x;
+extern uint _group_id_y;
+extern uint _group_id_z;
+
+extern uint _local_id_x;
+extern uint _local_id_y;
+extern uint _local_id_z;
+
+// extern uint _global_offset_x;
+// extern uint _global_offset_y;
+// extern uint _global_offset_z;
+
+size_t _CL_OVERLOADABLE
+get_global_id(unsigned int dimindx)
```

Comment:
I've done something like this now, with one `.ll` that wraps all the NVVM intrinsics with functions that are callable from C.